tests/chainer_tests/functions_tests/loss_tests/test_discriminative_margin_based_loss.py

```diff
@@ -0,0 +1,96 @@
+import unittest
+
+import numpy
+
+import chainer
+from chainer.backends import cuda
+from chainer import functions
+from chainer import testing
+from chainer.testing import attr
+
+
+@testing.parameterize(*(testing.product({
+    'delta_v': [0.5],
+    'delta_d': [5],
+    'alpha': [1],
+    'beta': [1],
+    'gamma': [0.001],
+    'norm': [1],
+    'result': [9709.743276179566]
+}) + testing.product({
+    'delta_v': [3],
+    'delta_d': [10],
+    'alpha': [0.1],
+    'beta': [0.1],
+    'gamma': [0.1],
+    'max_n_clusters': [2],
+    'norm': [2],
+    'result': [2140420.229810252]
+})))
+class TestDiscriminativeMarginBasedClusteringLoss(unittest.TestCase):
+
+    def setUp(self):
+        self.max_n_clusters = 5
+        self.batch = 5
+        self.width = 10
+        self.height = 10
+        shape = (self.batch, self.max_n_clusters,
+                 self.height, self.width)
+
+        input_arr = numpy.linspace(0, 100,
+                                   shape[0] * shape[1] *
+                                   shape[2] * shape[3])
+        self.input = input_arr.reshape(shape)
+        self.gt = numpy.linspace(100, 0,
+                                 shape[0] * shape[1] *
+                                 shape[2] * shape[3])
+        self.gt = self.gt.reshape(shape)
+        self.gt_obj = [1, 2, 3, 4, 5]
+        self.gt_obj_idx = [[0], [0, 2],
+                           [0, 3, 4], [0, 1, 2, 3],
+                           [0, 1, 2, 3, 4]]
+        self.y = numpy.asarray(self.result)
+
+    def get_result(self, x_data):
+        out = functions.discriminative_margin_based_clustering_loss(
+            x_data, self.delta_v, self.delta_d,
+            self.max_n_clusters, self.norm, self.alpha,
+            self.beta, self.gamma)
+        return out
+
+    def check_forward_cpu(self, x_data, t_data):
+        t = chainer.Variable(t_data)
+        out = self.get_result(x_data)
+        assert(out.data == t.data)
```

Comment:
I think it is better to admit some numerical errors by using e.g. [`numpy.testing.assert_almost_equal`](https://docs.scipy.org/doc/numpy/reference/generated/numpy.testing.assert_almost_equal.html)